app/src/main/java/de/tum/in/tumcampusapp/activities/CafeteriaActivity.java

```diff
@@ -142,16 +128,36 @@ public View getDropDownView(int position, View convertView, @NonNull ViewGroup p
                 return v;
             }
         };
-
         Spinner spinner = findViewById(R.id.spinnerToolbar);
         spinner.setAdapter(adapterCafeterias);
         spinner.setOnItemSelectedListener(this);
-        // Select item
-        if (selIndex > -1) {
-            spinner.setSelection(selIndex);
-        }
+        //TODO: handle on empty, cause right now we no longer handle no internet connection or empty results at all
+        //TODO: only select first item on activity start, but not on database update.
+        mDisposable.add(cafeteriaViewModel.getAllCafeteria(new LocationManager(this).getCurrentOrNextLocation())
+                                          .subscribe(
+                                                  it -> {
```

Comment:
Uggrh, this is hardly readable. Can we please not chain like 10 method calls? :s

Or is it just me? :sob:

Reply:
55 characters indentation seems a little bit much 🤔 

creating temporary variables or even breaking out a method for the subscriber probably would be nicer

Reply:
agree, that's why it's labeled wip 🙈